packages/yourloops/components/patient-list/current-patient-list/current-patient-list.hook.tsx

```diff
@@ -218,74 +212,9 @@ export const useCurrentPatientListHook = (props: CurrentPatientListHookProps): P
-      const birthdate = patient.profile.birthdate
-      return {
-        id: patient.userid,
-        [PatientListColumns.Flag]: patient,
-        [PatientListColumns.Patient]: patient,
-        [PatientListColumns.DateOfBirth]: patient,
-        [PatientListColumns.Age]: PatientUtils.computeAge(birthdate),
-        [PatientListColumns.Gender]: PatientUtils.getGenderLabel(patient.profile.sex),
-        [PatientListColumns.MonitoringAlerts]: patient,
-        [PatientListColumns.System]: patient.settings.system ?? trNA,
-        [PatientListColumns.LastDataUpdate]: lastUpload,
-        [PatientListColumns.Messages]: patient.metadata.hasSentUnreadMessages,
-        [PatientListColumns.TimeInRange]: patient.glycemiaIndicators.timeInRange,
-        [PatientListColumns.GlucoseManagementIndicator]: patient.glycemiaIndicators.glucoseManagementIndicator,
-        [PatientListColumns.Hypoglycemia]: patient.glycemiaIndicators.hypoglycemia,
-        [PatientListColumns.Variance]: patient.glycemiaIndicators.coefficientOfVariation,
-        [PatientListColumns.Actions]: patient
-      }
-    })
-  }, [patients, trNA])
-
-  const buildPrivateTeamOrCaregiverRows = useCallback((): GridRowsProp => {
-    return patients.map((patient): GridRowModel => {
-      const { lastUpload } = getMedicalValues(patient.metadata.medicalData, trNA)
-      const birthdate = patient.profile.birthdate
-      return {
-        id: patient.userid,
-        [PatientListColumns.Flag]: patient,
-        [PatientListColumns.Patient]: patient,
-        [PatientListColumns.DateOfBirth]: patient,
-        [PatientListColumns.Age]: PatientUtils.computeAge(birthdate),
-        [PatientListColumns.Gender]: PatientUtils.getGenderLabel(patient.profile.sex),
-        [PatientListColumns.System]: patient.settings.system ?? trNA,
-        [PatientListColumns.LastDataUpdate]: lastUpload,
-        [PatientListColumns.TimeInRange]: patient.glycemiaIndicators.timeInRange,
-        [PatientListColumns.GlucoseManagementIndicator]: patient.glycemiaIndicators.glucoseManagementIndicator,
-        [PatientListColumns.Hypoglycemia]: patient.glycemiaIndicators.hypoglycemia,
-        [PatientListColumns.Variance]: patient.glycemiaIndicators.coefficientOfVariation,
-        [PatientListColumns.Actions]: patient
-      }
-    })
-  }, [patients, trNA])
-
-  const rowsProps: GridRowsProp = useMemo(() => {
-    if (user.isUserCaregiver() || selectedTeam.type === TeamType.private) {
-      return buildPrivateTeamOrCaregiverRows()
-    }
-    return buildMedicalTeamRows()
-  }, [buildMedicalTeamRows, buildPrivateTeamOrCaregiverRows, selectedTeam, user])
-
-  return {
-    columns,
-    patientToRemoveForHcp,
-    patientToRemoveForCaregiver,
-    rowsProps,
-    onCloseRemoveDialog,
-    onRowClick
+  const onRowClick = (params: GridRowParams): void => {
+    navigate(`${AppUserRoute.Patient}/${params.id}/dashboard`)
   }
+
+  return { allRows, allColumns, onRowClick }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Another approach (you might have thought of it and did not like it) would have been to not have a generic hook returning every columns and rows, but to encapsulate columns objects into methods (defined in a util). Rows could have been duplicated (IMO there is no need to make them generic, but could still have been handled the same way as columns).

This would have had the advantage to not set non used values in object (and thereby not making some useless computing) (also, no need for lodash :open_mouth: ).

In a file such as `current-patient-list.util.tsx`:

```
 const buildFlagColumn = (): GridColDef => {
    return {
      field: PatientListColumns.Flag,
      type: 'actions',
      headerName: '',
      width: 55,
      hideable: false,
      sortable: true,
      sortComparator: sortByFlag,
      renderCell: (params: GridRenderCellParams<GridRowModel, Patient>): JSX.Element => {
        const patient = params.value
        return <FlagIconCell isFlagged={patient.metadata.flagged} patient={patient} />
      }
    }
  }

...

const buildActionColumn = (): GridColDef => {
    return {
      type: 'actions',
      field: PatientListColumns.Actions,
      headerName: t('actions'),
      headerClassName: classes.mandatoryCellBorder,
      cellClassName: classes.mandatoryCellBorder,
      renderCell: (params: GridRenderCellParams<GridRowModel, Patient>) => {
        return <ActionsCell patient={params.value} onClickRemove={onClickRemovePatient} />
      }
    }
  }
```

In `current-medical-team-patient-list.hook.tsx` and `current-private-team-or-caregiver-patient-list.hook.tsx`  
```
   const columns = [
    buildFlagColumn(),
   ...,
    buildActionColumn(onClickRemovePatient)
  ]
```